tests/unit/via/services/pdf_url_test.py

```diff
@@ -0,0 +1,136 @@
+from datetime import datetime, timedelta, timezone
+from unittest.mock import sentinel
+
+import pytest
+
+from via.services.pdf_url import PDFURLBuilder, _NGINXSigner, factory
+
+
+class TestNGINXSigner:
+    def test_it(
+        self,
+        svc,
+        quantized_expiry,
+    ):
+        signed_url = svc.sign_url(
+            "https://example.com/foo/bar.pdf?q=s", "/proxy/static/"
+        )
+
+        quantized_expiry.assert_called_once_with(max_age=timedelta(hours=25))
+        signed_url_parts = signed_url.split("/")
+        signature = signed_url_parts[5]
+        expiry = signed_url_parts[6]
+        assert signature == "qTq65RXvm6P2Y4bfzWdPzg"
+        assert expiry == "1581183021"
+
+    @pytest.fixture
+    def svc(self, pyramid_settings):
+        return _NGINXSigner(
+            pyramid_settings["nginx_server"],
+            pyramid_settings["nginx_secure_link_secret"],
+        )
+
+    @pytest.fixture(autouse=True)
+    def quantized_expiry(self, patch):
+        return patch(
+            "via.services.pdf_url.quantized_expiry",
+            return_value=datetime(
+                year=2020,
+                month=2,
+                day=8,
+                hour=17,
+                minute=30,
+                second=21,
+                tzinfo=timezone.utc,
+            ),
+        )
```

Comment:
This is code as you found it, so keeping it is absolutely fine. I don't think you have to make it any better just because it's related to something you're working on.

I had a few separate suggestions to make this shorter, but due to the hashing you can't really do that and have it work. In any case... 

```suggestion
    def test_it(self, quantized_expiry):
        service = _NGINXSigner(nginx_server="http://nginx", secret="not_a_secret")

        signed_url = service.sign_url("https://other/?q=s", "/path/")

        quantized_expiry.assert_called_once_with(max_age=timedelta(hours=25))
        assert (
            signed_url
            == "http://nginx/path/cYa67Tuf4V3j6KuQsAuBNw/1581120000/https://other/?q=s"
        )

    @pytest.fixture(autouse=True)
    def quantized_expiry(self, patch):
        return patch(
            "via.services.pdf_url.quantized_expiry",
            return_value=datetime(year=2020, month=2, day=8, tzinfo=timezone.utc),
        )
```